fetch-in-react/my-app/src/UserDetails.jsx

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import UserCard from './UserCard';
+
+export default function UserDetails({ userId, onCancel }) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState();
+  const [user, setUser] = useState();
+
+  /* your code here (hint: useEffect) */
+  useEffect(() => {
+    const response = new Request(`https://jsonplaceholder.typicode.com/users/${userId}`);
+    fetch(response)
+      .then((response) => {
+        return response.json();
+      })
+      .then((data) => {
+        setUser(data),
+        setIsLoading(false);
+      })
+      .catch((error) => {
+        setError(error),
+        setIsLoading(false);
+      })
+  });
```

Comment:
We only want this useEffect to run whenever the `userId` changes (aka we want it to run only when the user changes).